src/main/java/io/github/moulberry/notenoughupdates/miscgui/itemcustomization/ItemCustomizationUtils.java

```diff
@@ -0,0 +1,344 @@
+	}
+
+	public static String getChromaStrFromLeatherColour(GuiItemCustomize instance) {
+		ItemStack customItemStack = instance.customItemStack;
+		return ChromaColour.special(0, 0xff, ((ItemArmor) customItemStack.getItem()).getColor(customItemStack));
+	}
+
+	public static void renderFooter(int xCenter, int yTop, GuiType guiType) {
+		int xCentreLeft = xCenter - 90;
+		int xCentreRight = xCenter;
+
+		Gui.drawRect(xCentreLeft, yTop, xCenter + 1, yTop + 17, 0xff101016);
+		Gui.drawRect(xCentreLeft, yTop, xCenter - 1, yTop + 15, 0xff101016);
+		Gui.drawRect(xCentreLeft - 1, yTop + 1, xCenter - 2, yTop + 14, 0xff000000 | 0xff00ffc4);
+
+		Utils.renderShadowedString(getButtons(guiType, 0).getDisplay(),
+			xCentreLeft + 44,
+			yTop + 4,
+			xCenter * 2 - xCentreRight
+		);
+
+		xCentreLeft += 90;
+		xCentreRight += 90;
+
+		Gui.drawRect(xCentreLeft, yTop, xCentreRight, yTop + 17, 0x70000000);
+		Gui.drawRect(xCentreLeft, yTop, xCentreRight, yTop + 15, 0xff101016);
+		Gui.drawRect(xCentreLeft - 1, yTop + 1, xCentreRight, yTop + 14, 0xff000000 | 0xff00ffc4 * 2);
+
+		Utils.renderShadowedString(getButtons(guiType, 1).getDisplay(),
+			xCentreLeft + 45,
+			yTop + 4,
+			xCenter * 2 - xCentreRight
+		);
+	}
+
+	public static GuiType getButtons(GuiType guiType, int button) {
+		if (button == 0) {
+			if (guiType == GuiType.DEFAULT) {
+				return GuiType.ANIMATED;
+			} else {
+				return GuiType.DEFAULT;
+			}
+		}
+		if (button == 1) {
+			if (guiType == GuiType.HYPIXEL) {
+				return GuiType.ANIMATED;
+			} else {
+				return GuiType.HYPIXEL;
+			}
+		}
+
+		return GuiType.DEFAULT;
+	}
+
+	public static GuiType getButtonClicked(int mouseX, int mouseY, GuiType guiType, float offset) {
+		ScaledResolution scaledResolution = new ScaledResolution(Minecraft.getMinecraft());
+		int xCenter = scaledResolution.getScaledWidth() / 2;
+		int xCentreLeft = xCenter - 90;
+		int xCentreRight = xCenter;
+
+		for (int i = 0; i < 2; i++) {
+			if (mouseX >= xCentreLeft && mouseX <= xCentreRight &&
+				mouseY >= offset - 7 && mouseY <= offset + 12) {
+				return getButtons(guiType, i);
+			}
+			xCentreLeft += 90;
+			xCentreRight += 90;
+		}
+
+		return null;
+	}
+
+	public static int getAnimatedDyeColour(String[] dyeColours, int ticks) {
+		return ChromaColour.specialToChromaRGB(
+			dyeColours[(Minecraft.getMinecraft().thePlayer.ticksExisted / ticks) % dyeColours.length]);
+	}
+
+	static final ResourceLocation RESET = new ResourceLocation("notenoughupdates:itemcustomize/reset.png");
+
+	public static void renderColourBlob(int xCenter, int yTop, int colour, String text, boolean renderReset) {
+		Gui.drawRect(xCenter - 90, yTop, xCenter + 92, yTop + 17, 0x70000000);
+		Gui.drawRect(xCenter - 90, yTop, xCenter + 90, yTop + 15, 0xff101016);
+		Gui.drawRect(xCenter - 89, yTop + 1, xCenter + 89, yTop + 14, 0xff000000 | colour);
+
+		Utils.renderShadowedString(text, xCenter, yTop + 4, 180);
+
+		if (renderReset) {
```

Comment:
Not sure if we should use reset here and not something like an X for the multiple animated dyes.

Reply:
Ive asked june if she could make a texture for this